src/main/java/world/bentobox/bentobox/managers/IslandWorldManager.java

```diff
@@ -50,36 +50,48 @@ public IslandWorldManager(BentoBox plugin) {
         gameModes = new HashMap<>();
     }
 
-    public void registerWorldsToMultiverse() {
+    /**
+     * Registers or unregisters worlds with world management plugins
+     *
+     * @param reg true to register, false to remove registration
+     *
+     * Updated 1.24.0
+     */
+    public void registerWorldsToMultiverse(boolean reg) {
         gameModes.values().stream().distinct().forEach(gm -> {
-            registerToWorldManagementPlugins(gm.getOverWorld(), true);
+            registerToWorldManagementPlugins(gm.getOverWorld(), true, reg);
             if (gm.getWorldSettings().isNetherGenerate()) {
-                registerToWorldManagementPlugins(gm.getNetherWorld(), gm.getWorldSettings().isNetherIslands());
+                registerToWorldManagementPlugins(gm.getNetherWorld(), gm.getWorldSettings().isNetherIslands(), reg);
             }
             if (gm.getWorldSettings().isEndGenerate()) {
-                registerToWorldManagementPlugins(gm.getEndWorld(), gm.getWorldSettings().isEndIslands());
+                registerToWorldManagementPlugins(gm.getEndWorld(), gm.getWorldSettings().isEndIslands(), reg);
             }
         });
     }
 
-    /**
-     * Registers a world with world management plugins
-     *
-     * @param world the World to register
-     * @param islandWorld true if this is an island world
-     */
-    private void registerToWorldManagementPlugins(@NonNull World world, boolean islandWorld) {
-        if (plugin.getHooks() != null) {
-            for (Hook hook : plugin.getHooks().getHooks()) {
-                if (hook instanceof final WorldManagementHook worldManagementHook) {
-                    if (Bukkit.isPrimaryThread()) {
-                        worldManagementHook.registerWorld(world, islandWorld);
-                    } else {
-                        Bukkit.getScheduler().runTask(plugin, () -> worldManagementHook.registerWorld(world, islandWorld));
-                    }
+
+    private void registerToWorldManagementPlugins(@NonNull World world, boolean islandWorld, boolean reg) {
+        if (plugin.getHooks() == null) {
+            return;
+        }
+        for (Hook hook : plugin.getHooks().getHooks()) {
+            if (hook instanceof final WorldManagementHook worldManagementHook) {
+                if (Bukkit.isPrimaryThread()) {
+                    runTask(worldManagementHook, world, islandWorld, reg);
+                } else {
+                    Bukkit.getScheduler().runTask(plugin, () -> runTask(worldManagementHook, world, islandWorld, reg));
                 }
             }
         }
+
+    }
+
+    private void runTask(WorldManagementHook worldManagementHook, @NonNull World world, boolean islandWorld, boolean reg) {
+        if (reg) {
+            worldManagementHook.registerWorld(world, islandWorld);
+        } else {
+            worldManagementHook.unregisterWorld(world);
```

Comment:
Unregistration is required for MV, but not myWorlds therefore it doesn't do anything in the MyWorlds hook.